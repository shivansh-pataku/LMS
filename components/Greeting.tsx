import React from "react";

interface GreetingProps {
  name?: string;
}

const Greeting: React.FC<GreetingProps> = ({ name }) => {
  const displayName = name && name.trim() !== "" ? name : "Guest";
  return <h1>Hello, {displayName}!</h1>;
};

export default Greeting;
