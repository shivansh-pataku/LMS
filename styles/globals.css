@import url('https://fonts.googleapis.com/css2?family=Jost:wght@300;400;500;600;700&family=Commissioner:wght@400;500;600&family=Montserrat:wght@500;600;700&display=swap');

/* Jost
Montserrat
Commissioner
Open Sans
Atkinson Hyperlegible Next */


/* @import url('https://fonts.googleapis.com/css2?family=Jost:ital,wght@0,100..900;1,100..900&family=Oswald:wght@200..700&display=swap'); */
/* @import url('https://fonts.googleapis.com/css2?family=Commissioner:wght@100..900&family=Jost:ital,wght@0,100..900;1,100..900&family=Montserrat:ital,wght@0,100..900;1,100..900&family=Oswald:wght@200..700&family=Source+Sans+3:ital,wght@0,200..900;1,200..900&display=swap'); */
/* @import url('https://fonts.googleapis.com/css2?family=Jost:ital,wght@0,100..900;1,100..900&family=Montserrat:ital,wght@0,100..900;1,100..900&family=Oswald:wght@200..700&display=swap" rel="stylesheet'); */
/* @tailwind base;
@tailwind components;
@tailwind utilities;*/

/* @import url('https://fonts.googleapis.com/css2?family=Commissioner:wght@100..900&family=Jost:ital,wght@0,100..900;1,100..900&family=Montserrat:ital,wght@0,100..900;1,100..900&family=Oswald:wght@200..700&family=Source+Sans+3:ital,wght@0,200..900;1,200..900&display=swap');  */
/* @tailwind base;
@tailwind components;
@tailwind utilities; */

/* @import url('https://fonts.googleapis.com/css2?family=Commissioner:wght@100..900&family=Jost:ital,wght@0,100..900;1,100..900&family=Montserrat:ital,wght@0,100..900;1,100..900&family=Oswald:wght@200..700&family=Source+Sans+3:ital,wght@0,200..900;1,200..900&display=swap'); */


/* :root {
    margin: 0;
    font-family: "Jost", serif;
    font-optical-sizing: auto;
    font-weight: 200;
    font-style: normal;
    background-color: snow;
} */


/* html, body {
    height: 100%;
    overflow: hidden;
    Prevents whole page from scrolling
  } */
  
/* #root {
    max-width: 1280px;
    margin: 0 auto;
    padding: 2rem;
    text-align: center;
  } */

  


body {
    margin: 0;
    font-family: "Jost", serif;
    font-optical-sizing: auto;
    font-weight: 200;
    font-style: normal;
    background-color: snow;
    zoom: 80%;

}

html, body, #__next {
  height: 100%;
  margin: 0;
  padding: 0;
}

#major_container {
    display: flex;   /*  or grid */
 
    flex-wrap: wrap;
    justify-content: center;
    /* align-items: center; */
    width: 100%;
    padding: 20px;
    margin: auto;
    /* border: 2px solid black; */
    background-color: rgb(253, 253, 253);
    row-gap: 20px;
    margin-top: 100px;
    align-items: flex-start;

}

#major_container {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
    width: 110%;
    padding: 20px;
    margin: auto;
    /* border: 2px solid black; */
    background-color: rgb(253, 253, 253);
    row-gap: 20px;
    margin-top: 100px;
}

.container {
    width: auto;
    display:flex;
    justify-content:flex-start;
    /* flex-wrap: wrap; */
    /* grid-template-columns: repeat(2, 1fr); */
    /* Creates a responsive grid */
    gap : 20px;
    padding: 20px;
} 

.classic_heading{
    color: #184e10;
    font-family: Montserrat;
    font-weight: 670;
    font-size:xx-large;
    /* margin: 0px 0px 15px 15px; */
    padding: 0px 0px 0px 0px;
}

.classic_SUBheading{
    color: #184e10;
    font-family: jost;
    font-weight: 500;
    font-size: x-large;
    margin-top: 19px;
    margin-bottom: 12px;
    /* margin: 30px 0px 10px 0px; */
    padding: 0px 0px 0px 0px;
}

.row{
    width: 1011px;
    height: 25px;
    border-collapse: collapse;
    table-layout: auto;
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
    padding: 0;
    margin: 0;
    border: 1px solid #184e10;
    font-family: Commissioner;
    font-weight: 400;
    font-size: small;
    align-items: center;
    border-radius: 1px;
    align-content: space-between;
}


.GridButtons {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 4px 10px 4px 10px;
    padding: 8px 15px 8px 15px;
    padding: auto;
    width: 100%;
    height: 100%;
    background-color:rgba(21, 209, 21, 0.633);
    border: none;
    font-family: Commissioner;
    font-weight: 600;
    color: #09428dcd;
    border-radius: 3px;
}


.GridButtons:hover {
    background-color:rgba(12, 120, 12, 0.633);
    color: white;
    font-weight: 600;
    text-shadow: 0px 0px 8px #74747932;
    transition: all 0.3s ease;
}

.TV ,.TA{
    height: 17px;
    /* background-color: #184e1044; */
    text-align: left;
    padding: 3px;
    margin: 1px;
    width: 13%;
    font-family: Commissioner;
    font-weight: 400;
    /* border: 1px solid #184e10/; */
}

.TVE{
    height: 17px;
    /* background-color: #184e1044; */
    text-align: left;
    padding: 3px;
    margin: 1px;
    width: 13%;
    font-family: Commissioner;
    font-weight: 400;
    z-index: 1;
    background-color: rgba(108, 68, 75, 0.187);
    border-radius: 2px;
    border : 1px solid #184e10;
    border: none; 
}

.TV{
background-color: #184e1048;
font-weight: 500;
}

.TA{
background-color: #10264e62;
}

#error{
    height: 17px;
    font-family: Commissioner;
    font-weight: 500;
    text-decoration: none;
    color: #b94015;
    display: block;
    text-align: center;
}


#goto{
    height: 17px;
    font-family: Commissioner;
    font-weight: 500;
    text-decoration: none;
    color: #a77f07;
    display: block;
    text-align: center;
}
#goto:hover{
    text-decoration: none;
    font-weight: 600;
    text-shadow: 0px 0px 8px #74747932;
    transition: all 0.3s ease;
}

.ibCLASSIC {
    padding: 6px;
    border: .5px solid;
    font-family: Jost, serif;
    font-weight: 500;
    width: 163px;
    border-width: 1.5px;
    border-radius: 8.8px;
    box-sizing: border-box;
    background-color: #fff0;
    font-size: small;
    text-decoration-color: #000;
}


.JUSTbutton {
  align-items: center;
 
  background-color: #f6f6fe;
  border-radius: 4px;
  border-width: 0;
  box-shadow: 0 0px 2px #2d234266, 0 -2px 13px -11px #2d23424d, inset 0px 1px #d6d6e7;
  color: #36395A;
  cursor: pointer;
  display: inline-flex;
  font-family: jost;
  height: 40px;
  justify-content: center;
   list-style: none;
  margin: 5px 0px 0px 5px;
  padding: 0px 20px 0px 20px;
  position: relative;
    transition: box-shadow .15s,transform .15s;
  font-size: 18px;
}

.JUSTbutton:hover {
  box-shadow: rgba(45, 35, 66, 0.204) 0 4px 8px, rgba(45, 35, 66, 0.151) 0 7px 13px -3px, #D6D6E7 0 1px 0 inset;
  transform: translateY(-1px);
  color: rgb(15, 111, 13);
}

/* //////////////////////////////////////////////// NOT USED /////////////////////////////////////////////////  */


#a1{
    width: 200px;
}
#a2{
    width: 113px;
}
#a3{
    width: 55px;
}
#a4{
    width: 62px;
}
#a5{
    width: 60px;
}
#a6{
    width: 350px
}
#a7{
    width: 115px;
}


.submit {

    font-size: 14px;
    /* padding: .1cm; */
    display: block;
    /* margin: auto; */
    /* margin-top: 1px; */
    padding: 0px;
    font-family: Jost, serif;
    font-weight: 500;
    width: 123px;
    height: 23px;
    background-color: #14c125;
    border-width: 1px;
    border-radius: 1.8px;
    cursor: pointer;
}



/* /////////////////////////////////////////////////////// AG GRIDS CSS /////////////////////////////////////////////////////// */


.ag-header-cell {
    font-size: 14px;
    text-transform: uppercase;
    background-color: #00c40d9c;
}

/* Custom AG Grid Styling */
.ag-theme-alpine {
    font-family: comissioner;
    font-size: 14px;
    /* height: 50px; */
    height: 80px !important;
  }

  /* Header Styling */
  .ag-header {
    /* background-color: #007bff !important;  */
    color: white !important;
    font-weight: bold;
    text-align: center;
  }
  
  /* Column Headers */
  .ag-header-cell {
    /* height: 40px !important;
    line-height: 40px !important; */
    font-size: 14px;
    /* padding: 10px; */
    text-transform: uppercase;
    font-weight: 500 !important;
    /* border-right: 1px solid #ccc; */
    /* height: 50px; */
  }
  
  
  /* Row Styling */
  .ag-row {
    background-color: #f9f9f9; /* Light gray background */
  }
  
  /* Alternate Row Styling */
  .ag-row:nth-child(even) {
    background-color: #e9ecef; /* Slightly darker */
  }
  
  /* Selected Row Styling */
  .ag-row-selected {
    background-color: #ffeeba !important; /* Yellow highlight */
    color: black;
    font-weight: bold;
  }
  
  /* Editable Cell Styling */
  .ag-cell-editable {
    background-color: #d4edda !important; /* Light green */
  }
  
  /* Grid Borders */
  .ag-root-wrapper {
    border-radius: 10px;
    /* box-shadow: 2px 2px 10px rgba(0, 0, 0, 0.2); */
  }

/* //////////////hdfh//////hhdhdhh////hfghdhd///////////////////////////////// 
.ag-center-cols-viewport {
    min-height: unset !important;
}
  .ag-root-wrapper.ag-layout-normal {
    height: 70%;
    width: 100%;
    border-radius: 5px;}
  
  /* Grid Height and Width */
  .ag-theme-alpine {
    /* height: 350px  !important; */
    height:50%  !important;
    width: 100%;
    border-radius: 8px;
  }
  
  /* Centering Cell Content */
  .ag-cell {
    text-align: left;
    padding: auto;
        /* padding: 5px; */
    font-family: commissioner;
    color: rgb(18, 70, 106);
    font-weight: 500;
    font-size: small;

  }


  
  
element.style {
    left: 0px;
    width: 100px;
    height: 20px;

}

.custom-header {
    background-color: #00c40d9c !important; /* Orange */
    color: white !important;
    font-family: commissioner;
    height: 10px;
  }

.course-name{
    background-color: #00c40d9c !important; /* Orange */
    color: white !important;
    font-family: commissioner;
    /* height: 10px; */
  }


  /* /////////////////////////////////////// */

 
  /* Customize scrollbar */
  .ag-body-viewport::-webkit-scrollbar {
    width: 10px;
  }
  
  .ag-body-viewport::-webkit-scrollbar-track {
    background: #f1f1f1;
  }
  
  .ag-body-viewport::-webkit-scrollbar-thumb {
    background: #007bff;
    border-radius: 5px;
  }
  
  .ag-paging-panel {
    border-top: 1px solid;
    border-top-color: var(--ag-border-color);
    color: var(--ag-secondary-foreground-color);
    height: var(--ag-header-height);
    font-family: Commissioner;
    font-size: 15px;
    font-weight:400;
    background-color: bisque;
    height:fit-content; 

 
}

.ag-paging-panel > .ag-paging-page-size .ag-wrapper {
    min-width: calc(var(--ag-grid-size)* 10);
    border-radius: 10%;
    background-color: rgba(204, 157, 99, 0.355);

}

.GridButtons{
    margin: 10px 10px 10px 10px;
    padding: 8px 15px 8px 15px;
    cursor: pointer;
}



/* /////////////////////////////////////////////////////////////////////////// */

