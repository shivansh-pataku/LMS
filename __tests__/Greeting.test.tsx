import React from "react";
import { render, screen } from "@testing-library/react";
import Greeting from "@/components/Greeting"; // Assuming @/components alias is set up
import "@testing-library/jest-dom"; // Import jest-dom for extended matchers

// Remove 'expect' from this import to rely on the global Jest 'expect'
import { describe, test } from "@jest/globals";

describe("Greeting Component", () => {
  test('renders greeting with default name "Guest"', () => {
    render(<Greeting />);
    const headingElement = screen.getByRole("heading", {
      name: /Hello, Guest!/i,
    });
    expect(headingElement).toBeInTheDocument();
  });

  test("renders greeting with provided name", () => {
    const testName = "World";
    render(<Greeting name={testName} />);
    const headingElement = screen.getByRole("heading", {
      name: `Hello, ${testName}!`,
    });
    expect(headingElement).toBeInTheDocument();
  });

  test("renders greeting with default name for empty string name", () => {
    render(<Greeting name="" />);
    const headingElement = screen.getByRole("heading", {
      name: /Hello, Guest!/i, // Assuming it defaults to Guest for empty string
    });
    expect(headingElement).toBeInTheDocument();
  });

  test("matches snapshot with name", () => {
    const { asFragment } = render(<Greeting name="Snapshot" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("matches snapshot without name", () => {
    const { asFragment } = render(<Greeting />);
    expect(asFragment()).toMatchSnapshot();
  });
});
