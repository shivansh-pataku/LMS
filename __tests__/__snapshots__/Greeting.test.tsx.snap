// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting Component matches snapshot with name 1`] = `
<DocumentFragment>
  <h1>
    Hello, Snapshot!
  </h1>
</DocumentFragment>
`;

exports[`Greeting Component matches snapshot without name 1`] = `
<DocumentFragment>
  <h1>
    Hello, Guest!
  </h1>
</DocumentFragment>
`;
